cpp/src/arrow/filesystem/gcsfs.h

```diff
@@ -34,6 +35,13 @@ struct ARROW_EXPORT GcsOptions {
 
   std::string endpoint_override;
   std::string scheme;
+  // \brief Location to use for creating buckets.
```

Comment:
What kind of value can a location have? Is it a "region" like in S3?

Reply:
Yes, they are [regions](https://cloud.google.com/storage/docs/locations).  I went back and forth on whether to call this location or region, but chose location because it corresponded with the API.  I don't have a strong preference.  @coryan any thoughts on this?

Reply:
There are locations that are not regions:

https://cloud.google.com/storage/docs/locations#location-mr

Probably not of much interest for analytics workloads, but who knows.

Reply:
I've always thought of multi-regions as magic regions instead of a different concept.  Unfortunately, people do use multi-regions for analytics workloads and experience the associated pain points.